### PR TITLE
Issue/1153 support for source elements inside videos

### LINKED ISSUE #1153: Video element with source child elements are not being parsed properly
**Describe the bug**

When parsing HTML content that has a video element that defines its media source using the source element the video is not correctly parsed.

**To Reproduce**
Steps to reproduce the behaviour:
1. Open  the editor and switch to HTML mode
2. Write the following HTML: 
```
<video>
    <source src="https://videos.files.wordpress.com/kUJmAcSf/bbb_sunflower_1080p_30fps_normal.mp4" type="video/mp4">
</video>
```
3. Switch to visual mode
4. See that the video is not displayed and the source elements show as an unsupported element

**Expected behavior**
The video element should be parsed and the video displayed.

**Screenshots**
video-source-bug

**Smartphone (please complete the following information):**
 - Device: iPhoneXs
 - OS: iOS12
 - Version: 1.4.4

### PR DESCRIPTION
**Pull request by SergioEstevao**
Fixes #1153 

This PR improves the parsing and serialization of video elements to support the reading and writing of source elements inside the video elements.

To test:
 - Open the demo app
 - Run the video demo
 - Check that all videos display correctly
 - Switch to HTML mode and check if HTML is still correct
 - Switch to Visual mode and check if video are correct
 - Tap on the last video, and see if video preview works correctly

### CODE REVIEW

File: Example/Example/SampleContent/video.html [hunk 1]
```diff
@@ -4,6 +4,14 @@ <h1>Standard Video</h1>
...
-[video src="https://videos.files.wordpress.com/kUJmAcSf/bbb_sunflower_1080p_30fps_normal.mp4" poster="https://videos.files.wordpress.com/kUJmAcSf/bbb_sunflower_1080p_30fps_normal_scruberthumbnail_2.jpg" alt="Another video with bunnies"/]
+<h1>Standard Video with source element</h1>
+<video alt="Another video with bunnies">
+    <source src="https://videos.files.wordpress.com/kUJmAcSf/bbb_sunflower_1080p_30fps_normal.mp4" type="video/mp4">
+</video>
+
+<h1>Standard Video with multiple source elements</h1>
+<video alt="Another video with bunnies">
+    <source src="https://videos.files.wordpress.com/kUJmAcSf/bbb_sunflower_1080p_30fps_normal.mp4" type="video/mp4">
+    <source src="https://videos.files.wordpress.com/kUJmAcSf/bbb_sunflower_1080p_30fps_normal_fmt1.ogv" type="video/ogg">
```
Comment (etoledom) [inline Example/Example/SampleContent/video.html]: I noticed that in the app this second source is removed:

```html
<h1>Standard Video with multiple source elements</h1>
<p><video alt="Another video with bunnies"><source src="https://videos.files.wordpress.com/kUJmAcSf/bbb_sunflower_1080p_30fps_normal.mp4" type="video/mp4"></video></p>
```

Comment (SergioEstevao) [reply Example/Example/SampleContent/video.html]: Good catch, I updated the code to fix this!

**Comment by SergioEstevao** [comment]
@etoledom thanks for your excellent tests! I addressed both your comments and the PR is ready for another look.

File: Aztec/Classes/Converters/ElementsToAttributedString/Implementations/VideoElementConverter.swift [hunk 2]
```diff
@@ -52,10 +52,37 @@ class VideoElementConverter: AttachmentElementConverter {
...
+        let sources = extractSources(from: element)
+
+        let attachment = VideoAttachment(identifier: UUID().uuidString, srcURL: srcURL, posterURL: posterURL, sources: sources)
 
         attachment.extraAttributes = extraAttributes
 
         return attachment
     }
+
+    func extractSources(from element: ElementNode) -> [VideoSource] {
```
Comment (frosty) [inline Aztec/Classes/Converters/ElementsToAttributedString/Implementations/VideoElementConverter.swift]: Should we perhaps just add a doc comment explaining what these methods do / why they're necessary?

**Pull request merged**